src/terraform.yml

```diff
@@ -39,14 +39,16 @@ repository:
       - include: "#expressions"
 
   functions:
-    begin: (\w+)(\()
+    begin: (\w+|(?:provider::\w+::\w+))(\()
```

Comment:
I'm not sure if it's beneficial to be extra explicit here. The regex in line 50 is different from the one used here and accepts `-`, while this one doesn't. If we want to include the provider prefix here, we need to keep both regexs in sync. 

I would prefer a more generic one here and let the one below do the heavy lifting.
e.g. `([:\-\w]+)(\()`